src/azul/service/avro_pfb.py

```diff
@@ -81,42 +91,52 @@ class PFBConverter:
     Relations.
     """
 
-    def __init__(self, schema: JSON):
+    def __init__(self, schema: JSON, repository_plugin: RepositoryPlugin):
         self.schema = schema
+        self.repository_plugin = repository_plugin
         self._entities: MutableMapping[PFBEntity,
                                        MutableSet[PFBRelation]] = defaultdict(set)
 
     def add_doc(self, doc: JSON):
         """
         Add an Elasticsearch document to be transformed.
         """
-        contents = doc['contents']
+        doc_copy = copy_json(doc, 'contents')
+        contents = doc_copy['contents']
```

Comment:
```suggestion
        contents = doc_copy['contents', 'files']
```

and `contents['files'][:] = ...` below

That makes the copy much smaller.